drivers/gpio/gpio_pcal64xxa.c

```diff
@@ -185,14 +185,17 @@ static int pcal64xxa_process_input(const struct device *dev, gpio_port_value_t *
 	int rc;
 	uint16_t int_sources;
 	uint16_t input_port;
+	uint16_t fired_triggers = 0;
+	uint16_t dual_edge_triggers = drv_data->triggers.dual_edge;
```

Comment:
IMO `triggers` driver data should be accessed in the protected code segment as well as it can be modified asynchronously by `triggers_apply()`.